packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+
+| Event          | Signature                                                  | Description                                                     |
+| -------------- | ---------------------------------------------------------- | --------------------------------------------------------------- |
+| onCreate       | `(Object: newItem) => {}`                                  | Fired when the User saves a new item                            |
+| onUpdate       | `(Array[Object]: updatedItems) => {}`                      | Fired when one or more rows have been updated by the User       |
+| onDelete       | `(Array[String]: selectedItems) => {}`                     | Fired when one or more rows have been deleted by the User       |
+| onCustomAction | `(String: actionType, Array[String]: selectedItems) => {}` | Fired when a custom action has been applied to one or more rows |
+
+### Config
+
+There will be use cases where features like multi-selection, CRUD, custom actions, etc. won't be required at all, they must be disabled by thefault but easily activated. It makes sense to have boolean props that controll such features.
+
+Another important aspect of configuration is the definition of columns, hence a complete section is devoted to it.
+
+| Name    | Type          | default | Description                                        |
+| ------- | ------------- | ------- | -------------------------------------------------- |
+| Bulk    | Boolean       | `FALSE` | If true, activates the bulk mode (multi-select)    |
+| Columns | Array[Object] | `[]`    | Holds the definitions for each column in the table |
+| Create  | Boolean       | `FALSE` | If true, activates the creation of new items       |
+| Delete  | Boolean       | `FALSE` | If true, activates the deletion of items           |
+| Update  | Boolean       | `FALSE` | If true, activates the edit mode                   |
+
+### Triggers
+
+Custom actions are present in most tables, they are an important aspect of all applications. The nature of the actions depends on the entity they are applied to, hence, an abstract way of defining custom actions is required.
+
+| Attribute | Type     | default      | Description                                                 |
+| --------- | -------- | ------------ | ----------------------------------------------------------- |
+| id        | String   | `''`         | The id of the action, usually the name.                     |
+| callback  | Function | `noop`       | The function to be called when the action button is clicked |
+| trigger   | ANY      | `<Button />` | The element that will trigger the action                    |
+
+### Data
+
+The data passed to the table for rendering purposes.
+
+| Name | Type          | default | Description                                   |
+| ---- | ------------- | ------- | --------------------------------------------- |
+| data | Array[Object] | `[]`    | the tabular data to be displayed by the table |
```

Comment:
I wonder if there are some consistent things we could at least make sure are present in these data objects. I guess the only thing for sure that we would want to make sure of is `eid` which could then be used for keys in the table view. Past that, these objects could look like almost anything.

An `eid` will be super important for data arrays on anything virtualized too.

Not sure if we should necessarily address that here. 

Also, like the pattern of CRUD operations, I think that a `data` prop is also going to be some of our bread and butter in Cascara. We should also consider a section for making notes of how this particular prop should be used too.